src/vsr/replica.zig

```diff
@@ -3075,57 +3111,28 @@ pub fn ReplicaType(
             assert(self.status == .normal);
             assert(self.backup());
             assert(self.repair_sync_timeout.ticking);
-
             self.repair_sync_timeout.reset();
 
-            if (self.syncing != .idle) return;
-            // May as well wait for an in-progress checkpoint to complete —
-            // we would need to wait for it before sync starts anyhow, and the newer
-            // checkpoint might sidestep the need for sync anyhow.
-            if (self.commit_stage == .checkpoint_superblock) return;
-            if (self.commit_stage == .checkpoint_data) return;
-
-            // TODO Test connectivity to cluster to rule out a network partition.
-
-            // Even if we want to transition to sync, there is no point if we don't have
-            // a target yet (or if our latest target would not advance our state).
-            if (self.sync_target_max == null) return;
-            if (self.sync_target_max.?.checkpoint_op <= self.op_checkpoint()) return;
-
-            {
-                const commit_next = self.commit_min + 1;
-                const commit_next_slot = self.journal.slot_with_op(commit_next);
-
-                // "stuck" is not actually certain, merely likely (see below).
-                const stuck_header = !self.valid_hash_chain(@src());
-
-                const stuck_prepare =
-                    (commit_next_slot == null or self.journal.dirty.bit(commit_next_slot.?));
-
-                const stuck_grid = !self.grid.read_global_queue.empty();
-
-                if (!stuck_header and !stuck_prepare and !stuck_grid) return;
+            if (self.syncing == .awaiting_checkpoint or self.repair_stuck()) {
+                log.warn("{}: on_repair_sync_timeout: request sync; lagging behind cluster " ++
+                    "(op_head={} commit_min={} commit_max={} commit_stage={s})", .{
+                    self.replica,
+                    self.op,
+                    self.commit_min,
+                    self.commit_max,
+                    @tagName(self.commit_stage),
+                });
+                self.send_header_to_replica(
+                    self.primary_index(self.view),
+                    @bitCast(Header.RequestStartView{
```

Comment:
Unlike the old sync flow, we now don't preserve the state which tells us "we were stuck when the timeout fired" anymore. Rather, the state is preseved implictly by the network, when we send request start view, and then, when SV arrives to us, we just go to state sync.

In other words, it is possible that the priary decides to send and SV to us out of the blue, and, if we receive it and _at that point_ `repair_stuck` returns true, we'll go into the state sync, even though we weren't a lot of time in the repair_stuck state.

But I don't think we are sending SVs at random, and, if we receive one, it is _most_ likely due to us sending a request beforehand